compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -817,7 +821,19 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             return Ok(cycle_result);
         }
 
-        let (result, dep_node) = self.in_task(|this| this.evaluate_stack(&stack));
+        let (result, dep_node) = if cfg!(parallel_compiler) {
```

Comment:
Why not use this branch in both parallel and serial code?

Reply:
Just wanted to minimize the impact on serial mode. We should indeed unify here for code brevity.